src/Convert.h

```diff
@@ -1,11 +1,93 @@
 #ifndef CONVERT_H
 #define CONVERT_H
 
-class QString;
+#include <QString>
+#include <cstdint>
+#include <optional>
+#include <limits>
 
-int stringToValue(const QString& str);
 QString hexValue(int value, int width = 0);
 QString escapeXML(QString str);
 QString unescapeXML(QString str);
 
+template <typename T>
+std::optional<T> make_positive_optional(int value)
+{
+    return value < 0 ? std::nullopt : std::optional(value);
+}
+
+// Create optional<T> if boolean b is true.
+template <typename T, typename X>
+std::optional<T> make_if(X b, T value)
+{
+	static_assert(std::is_constructible<bool, X>::value, "X is not convertible to bool");
+	return b ? value : std::optional<T>();
+}
+
+// Create optional<T> if boolean b is true (solves implicit optional in T).
+template <typename T, typename X>
+std::optional<T> make_if(X b, std::optional<T> value)
+{
+	static_assert(std::is_constructible<bool, X>::value, "X is not convertible to bool");
+	return b ? (value ? *value : std::optional<T>()) : std::optional<T>();
+}
```

Comment:
> Why do you make the first parameter type a template and then in the implementation enforce that it must be convertible to bool? Wouldn't it be simpler to take a bool parameter? (And in case of the template version, you likely want to take the parameter by const-reference).

Because I intended to use it primarily to check if a optional is valid with something like `make_if(bp.slot.ps, ...)`, but the function wouldn't consider `bp.slot.ps` a boolean, returning this error:
```
error: no matching function for call to ‘make_if(std::optional<signed char>&, std::optional<signed char>&)’

src/Convert.h:21:18: note:   template argument deduction/substitution failed:
src/BreakpointViewer.cpp:327:53: note:   cannot convert '' (type ‘std::optional<signed char>’) to type ‘const bool&’
(...)
```
So I used template type deduction. But it was not useful after all because I forgot that it would not short-circuit if condition is `false`. So I am considering writing a lambda version of it, like this:
`
int8_t slotCmbIdx = make_if(bp.slot, [&]{ return *bp.slot.ps + 1; });
`
So `return bp.slot.ps` will only execute if `bp.slot.ps` evaluates to `true`.